lib/modules/manager/rust/extract.ts

```diff
@@ -0,0 +1,19 @@
+import { logger } from '../../../logger';
+import { GithubReleasesDatasource } from '../../datasource/github-releases';
+import type { PackageDependency, PackageFileContent } from '../types';
+
+export function extractPackageFile(content: string): PackageFileContent {
+  logger.trace('rust.extractPackageFile()');
+  const dep: PackageDependency = {
+    depName: 'rust',
+    packageName: 'rust-lang/rust',
+    currentValue: content.trim(),
+    datasource: GithubReleasesDatasource.id,
```

Comment:
Does the rust toolchain install directly from github releases? e.g. a second after a new release is published on github.com, it's possible to install it using rust toolchain?

Reply:
@kyle-leonhard As far as rustup is concerned, we use https://static.rust-lang.org/dist/rust-1.80-beta-aarch64-apple-darwin.tar.gz.sha256 as the single source of truth (replace `1.80-beta` and `aarch64` according to your situation.

Reply:
@kyle-leonhard As a matter of fact, we do GitHub tagging after other steps (https://github.com/rust-lang/promote-release/blob/110ad45f13ac5f0aa660acc218c090c6b24f14c8/src/main.rs#L246-L263), so I guess it's a reasonable reference to be used here as well.